### DISCUSSION TITLE
Docker-Compose Scaling Example

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [directus/directus#22505] [Feature Requests]
### Summary

As I can’t find a documentation on how to scale my Directus instance for production using docker I would really appreciate if there would be a minimalistic tutorial.

### Motivation

Due to starting the production Server it would be nice if Directus can handle this.

### Detailed Design

-

### Requirements List

-

### Drawbacks

-

### Alternatives

-

### Adoption Strategy

-

**Comment by br41nslug** [comment]
I feel this is somewhat outside the scope of Directus itself since there are various methods of hosting and scaling dockers with their own ways of working and drawbacks none of which are unique to Directus. Could you add some more detail to this like what technologies / architectures you'd like to see such a guide for?

Or do you mean a short guide on which environment variables matter once you have set up a horizontally, vertically or both scaled setup?

**Reply by NilsBaumgartner1994** [reply]
I mean more like, does Directus synchronize their workers somewhere like over Redis as having multiple instances would bring problems when dealing with flows?

Could I just scale Directus horizontally with —scale in docker-compose or with swarm where intell: replicas=7 or so?

**Reply by br41nslug** [reply]
For horizontal scaling a redis instance is required to synchronize events / crons / and such
https://docs.directus.io/self-hosted/config-options.html#synchronization

**Reply by NilsBaumgartner1994** [reply]
Good to know :-) Any other information on how to horizontal scale directus?

**Reply by NilsBaumgartner1994** [reply]
I found uncomplete information on Restack on that. Is restack associated with Directus?
https://www.restack.io/docs/directus-knowledge-directus-self-hosted-guide#clpk20u3v09d1x40v2z7hvkwe

**Reply by br41nslug** [reply]
Restack is not associated with Directus afaik

**Comment by NilsBaumgartner1994** [comment]
Sorry for late response. So technically, what I think would be nie to see: A basic docker example on how to use Directus in replicated mode (easy), simple mod balancer (traefik? Easy), and how to scale the database (hard? Maybe cockroachDb).

As this could help users to start going from development into production with their Directus project.
For my case we are currently perfect for handling around 40,000 monthly active users but we would like to scale up.

**Comment by NilsBaumgartner1994** [comment]
So the Directus docs on Kubernetes tried to show the idea what I mean https://docs.directus.io/blog/understanding-kubernetes.html

But it lacks the points:

How to scale the database, replicas?

And the loadbalancing

**Reply by SadmanYasar** [reply]
> So the Directus docs on Kubernetes tried to show the idea what I mean https://docs.directus.io/blog/understanding-kubernetes.html
> 
> But it lacks the points:
> 
> How to scale the database, replicas?
> 
> And the loadbalancing

Were you able to find a solution? I think there are still lack of docs for scaling Directus with docker compose.

**Reply by NilsBaumgartner1994** [reply]
I found a solution with docker compose scaling Directus

**Reply by NilsBaumgartner1994** [reply]
‘’’

services:

  # The Directus backend
  rocket-meals-directus:
    build:
      context: ./
    #container_name: "rocket-meals-directus" # "replicas" is ignored if you name your container using container_name: myname
    restart: ${RESTART}
    deploy:
      mode: replicated
      replicas: ${DIRECTUS_REPLICAS:-1}  # Default to 1 if DIRECTUS_REPLICAS is not set
      endpoint_mode: vip # Virtual IP - https://docs.docker.com/compose/compose-file/deploy/#endpoint_mode
    networks:
      - traefik
      - directus_network

‘’’


And then made the routing with traefik
